packages/webpack/src/runtime/runtimeBuilder.js

```diff
@@ -16,191 +60,335 @@ module.exports = {
+     * @returns {string} The assembled runtime source code
+     */
+    function getLavaMoatRuntimeSource({
+      embeddedOptions,
+      chunkIds,
+      policyData,
+      identifiers: {
+        root,
+        identifiersForModuleIds,
+        unenforceableModuleIds,
+        contextModuleIds,
+        externals,
       },
+    }) {
+      let repairs
+      if (options.skipRepairs === true) {
+        repairs = ''
+      } else {
+        repairs = require('./repairsBuilder.js').buildRepairs(
+          policyData,
+          options.skipRepairs
+        )
+      }
+
+      /** @type {RuntimeFragment[]} */
+      const runtimeFragments = [
+        // the string used to indicate root resource id
+        {
+          name: 'root',
+          data: root,
+          json: true,
+        },
+        // a mapping used to look up resource ids by module id
+        {
+          name: 'idmap',
+          data: identifiersForModuleIds,
+          json: true,
+        },
+        // list of ids of modules to skip in policy enforcement
+        {
+          name: 'unenforceable',
+          data: unenforceableModuleIds,
+          json: true,
+        },
+        // list of known context modules
+        {
+          name: 'ctxm',
+          data: contextModuleIds || null,
+          json: true,
+        },
+        // known chunk ids
+        {
+          name: 'kch',
+          data: chunkIds,
+          json: true,
+        },
+        // a record of module ids that are externals and need to be enforced as builtins
+        {
+          name: 'externals',
+          data: externals || null,
+          json: true,
+        },
+        // options to turn on scuttling
+        { name: 'options', data: embeddedOptions, json: true },
+        // scuttling module, if needed
+        (typeof embeddedOptions?.scuttleGlobalThis === 'boolean' &&
+          embeddedOptions.scuttleGlobalThis === true) ||
+        (typeof embeddedOptions?.scuttleGlobalThis === 'object' &&
+          embeddedOptions.scuttleGlobalThis.enabled === true)
+          ? {
+              name: 'scuttling',
+              shimRequire: 'lavamoat-core/src/scuttle.js',
+            }
+          : {},
+        // the policy itself
+        { name: 'policy', data: policyData, json: true },
+        // enum for keys to match the generated ones in wrapper
+        {
+          name: 'ENUM',
+          file: require.resolve('../ENUM.json'),
+          json: true,
+        },
+        // endowments module
+        {
+          name: 'endowmentsToolkit',
+          shimRequire: 'lavamoat-core/src/endowmentsToolkit.js',
+        },
+        // repairs
+        {
+          name: 'repairs',
+          rawSource: repairs,
+        },
+        // main lavamoat runtime
+        {
+          name: 'runtime',
+          file: require.resolve('./runtime.js'),
+        },
+      ]
+
+      if (options.debugRuntime) {
+        // optional debug helpers
+        runtimeFragments.push({
```

Comment:
(you'd have to do this inline, though)

Reply:
It makes things more cognitively complex to read but somehow still nicer.